var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -23,27 +24,28 @@ class PyNumba(PythonPackage):
     version("0.48.0", sha256="9d21bc77e67006b5723052840c88cc59248e079a907cc68f1a1a264e1eaba017")
     version("0.40.1", sha256="52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33")
 
-    depends_on("python@3.7:3.10", when="@0.55.0:", type=("build", "run"))
+    depends_on("python@3.8:3.12", when="@0.57", type=("build", "run"))
+    depends_on("python@3.7:3.11", when="@0.56", type=("build", "run"))
+    depends_on("python@3.7:3.10", when="@0.55", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("python@3.8:3.11", when="@0.57", type=("build", "run"))
    depends_on("python@3.7:3.10", when="@0.55:0.56", type=("build", "run"))
```
The upper bounds are exclusive (see [setup.py#L45](https://github.com/numba/numba/blob/0.55.2/setup.py#L45))